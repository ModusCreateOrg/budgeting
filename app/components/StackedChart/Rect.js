// @flow
import * as React from 'react';
import { select, interpolate } from 'd3';

type RectProps = {
  width: number,
  height: number,
  fill: string,
  y: number,
  animDuration: number,
};

class Rect extends React.Component<RectProps> {
  static defaultProps = {
    animDuration: 1000,
  };

  componentDidMount() {
    const { animDuration } = this.props;

    const rect = select(this.rectRef);

    if (this.rectRef) {
      const interpolateHeight = interpolate(1000, this.rectRef.getAttribute('height'));

      rect
        .transition()
        .duration(animDuration * Math.random())
        .attrTween('height', () => t => interpolateHeight(t));
    }
  }

  rectRef: ?HTMLElement;

  handleRefUpdate = (ref: ?HTMLElement) => {
    this.rectRef = ref;
  };

  render() {
    const { width, height, fill, y } = this.props;
    return <rect ref={this.handleRefUpdate} {...{ y, height, width, fill }} />;
  }
}

export default Rect;
