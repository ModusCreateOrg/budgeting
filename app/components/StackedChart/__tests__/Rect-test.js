import React from 'react';
import renderer from 'react-test-renderer';
import Rect from '../Rect';

jest.mock('d3');

it('renders correctly', () => {
  const tree = renderer.create(<Rect width={100} height={100} fill="test" y={3} />).toJSON();
  expect(tree).toMatchSnapshot();
});
