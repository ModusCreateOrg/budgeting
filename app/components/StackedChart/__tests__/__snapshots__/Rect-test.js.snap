// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<rect
  fill="test"
  height={100}
  width={100}
  y={3}
/>
`;
